lib/handlers/allow.js

```diff
@@ -93,56 +102,69 @@ function fetchDocument (host, ldp, baseUri) {
   }
 }
 
-function getUserId (req, callback) {
-  var onBehalfOf = req.get('On-Behalf-Of')
-  if (!onBehalfOf) {
-    return callback(null, req.session.userId)
+/**
+ * Determines the id of the current user from one of the following:
+ *
+ * - From `On-Behalf-Of` header (if it's present AND secretary is authorized)
+ * - From `session.userId` (if no delegation header, or secretary unauthorized)
+ *
+ * @method getUserId
+ * @param webId {string} Current user, from session.userId
+ * @param onBehalfOf {string} Contents of the `On-Behalf-Of` delegation header
+ * @param fetchDocument {Function} Result of `documentFetcher()` call
+ * @return {Promise<string>} The user id (either the current user, or
+ *   the user being delegated for).
+ */
+function getUserId (webId, onBehalfOf, fetchDocument) {
```

Comment:
I think this function could use a slightly more specific name. Something like `getRequestingWebId`?

Another thing is that I think the `webId` and `onBehalfOf` parameters are an implementation concern of this function rather than the caller. Could we just pass the request object instead?